controller-manager is in CrashLoopBackOff

```sh
[vagrant@metal3 go]$ kubectl get pods -n metal3
NAME                                                  READY   STATUS             RESTARTS   AGE
cluster-api-controller-manager-0                      1/1     Running            0          51m
cluster-api-provider-baremetal-controller-manager-0   1/2     CrashLoopBackOff   12         51m
metal3-baremetal-operator-6ffb74dfbb-864f6            3/3     Running            0          51m

[vagrant@metal3 go]$ kubectl logs -n metal3        cluster-api-provider-baremetal-controller-manager-0 -c manager
{"level":"info","ts":1559870504.1104558,"logger":"baremetal-controller-manager","msg":"Found API group metal3.io/v1alpha1"}
{"level":"info","ts":1559870504.8181705,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"machine-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1559870505.0203254,"logger":"kubebuilder.controller","msg":"Starting Controller","controller":"machine-controller"}
{"level":"info","ts":1559870505.1231196,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"machine-controller","worker count":1}
2019/06/07 01:21:45 Checking if machine centos exists.
2019/06/07 01:21:45 Machine centos exists.
2019/06/07 01:21:45 Updating machine centos .
E0607 01:21:45.325251       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:388
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator.go:459
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator.go:421
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator.go:187
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:208
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xebd65f]

goroutine 125 [running]:
github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x1122520, 0x1bd1470)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine.(*Actuator).nodeAddresses(0xc4203596f0, 0xc4200b3200, 0xc4200a60f0, 0x2, 0xc420301240, 0x19, 0x0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator.go:459 +0x5f
github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine.(*Actuator).updateMachineStatus(0xc4203596f0, 0x13953a0, 0xc4200480f0, 0xc42037edc0, 0xc4200b3200, 0x0, 0x0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator.go:421 +0x39
github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine.(*Actuator).Update(0xc4203596f0, 0x13953a0, 0xc4200480f0, 0x0, 0xc42037edc0, 0x1, 0x0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator.go:187 +0x1d3
github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc420524b80, 0xc42035be00, 0x6, 0xc42035bdf0, 0x6, 0x1be40e0, 0x0, 0x0, 0x0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:208 +0x6e1
github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc4200e4000, 0x0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x188
github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc42024efc0)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc42024efc0, 0x3b9aca00, 0x0, 0x1, 0xc4202c8780)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc42024efc0, 0x3b9aca00, 0xc4202c8780)
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/github.com/metal3-io/cluster-api-provider-baremetal/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x3
```

--- a/pkg/cloud/baremetal/actuators/machine/actuator.go
+++ b/pkg/cloud/baremetal/actuators/machine/actuator.go
@@ -451,8 +451,8 @@ func (a *Actuator) applyMachineStatus(ctx context.Context, machine *machinev1.Ma
 func (a *Actuator) nodeAddresses(host *bmh.BareMetalHost) ([]corev1.NodeAddress, error) {
 	addrs := []corev1.NodeAddress{}
 
-	// If the host is nil, return an empty address array.
-	if host == nil {
+	// If the host is nil or we have no hw details, return an empty address array.
+	if host == nil || host.Status.HardwareDetails == nil {
 		return addrs, nil
 	}
 

--- a/pkg/cloud/baremetal/actuators/machine/actuator_test.go
+++ b/pkg/cloud/baremetal/actuators/machine/actuator_test.go
@@ -897,6 +897,20 @@ func TestUpdateMachineStatus(t *testing.T) {
 				},
 			},
 		},
+		{
+			// machine status unchanged
+			Host: &bmh.BareMetalHost{},
+			Machine: &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mymachine",
+					Namespace: "myns",
+				},
+				Status: machinev1.MachineStatus{},
+			},
+			ExpectedMachine: machinev1.Machine{
+				Status: machinev1.MachineStatus{},
+			},
+		},
 	}
 
 	for _, tc := range testCases {

